salt/_modules/metalk8s_monitoring.py

```diff
@@ -0,0 +1,173 @@
+                "Malformatted response returned from Alertmanager API: {}: {}"
+                .format(str(exc), response.text)
+            )
+        raise CommandExecutionError(error)
+
+    if json['status'] == 'error':
+        raise CommandExecutionError(
+            "{}: {}".format(json['errorType'], json['error'])
+        )
+
+    return json.get('data')
+
+
+def add_silence(value, name='alertname', is_regex=False, starts_at=None,
+                duration=3600, ends_at=None, time_format='%Y-%m-%dT%H:%M:%S',
+                author='', comment=''):
+    """Add a new silence in Alertmanager.
+
+    Arguments:
+        value (str): value to check
+        name (str): label to check
+        is_regex (bool): Whether `value` should be treated as a regular
+            expression or not, defaults to False.
+        starts_at (str): Date when the silence starts, defaults to `now`.
+        duration (int): Duration of the silence in seconds, defaults to `3600`.
+        ends_at (str): Date when the silence ends, defaults to
+            `starts_at` + `duration`.
+        time_format (str): Time format for `starts_at` and `ends_at` arguments.
+            Support the `datetime` Python library flags.
+        author (str): Creator of the silence.
+        comment (str): A description of why this silence has been put.
+    """
+    if starts_at is None:
+        starts_at = datetime.now()
+    else:
+        starts_at = datetime.strptime(starts_at, time_format)
+
+    if ends_at is None:
+        ends_at = starts_at + timedelta(seconds=duration)
+    else:
+        ends_at = datetime.strptime(ends_at, time_format)
+
+    body = {
+        "matchers": [{
+            "name": name,
+            "isRegex": is_regex,
+            "value": value,
+        }],
+        "startsAt": starts_at.strftime('%Y-%m-%dT%H:%M:%SZ'),
+        "endsAt": ends_at.strftime('%Y-%m-%dT%H:%M:%SZ'),
+        "createdBy": author,
+        "comment": comment,
+        "status": {"state": "active"},
+    }
+
+    response = _requests_alertmanager_api(
+        'api/v1/silences',
+        'POST',
+        json=body,
+    )
+
+    return response['silenceId']
+
+
+def delete_silence(silence_id):
+    """Delete a silence in Alertmanager
+
+    Arguments:
+        silence_id (str): ID of the silence to delete.
+    """
+    _requests_alertmanager_api(
+        'api/v1/silence/{}'.format(silence_id),
+        'DELETE',
+    )
+
+
+def get_silences(state=None):
+    """Get the list of all silences in Alertmanager
+
+    Arguments:
+        state (str): Filter silences on their state (e.g. `active`),
+            if None, return all silences, defaults to `None`.
+    """
+    response = _requests_alertmanager_api(
+        'api/v1/silences',
+        'GET',
+    )
+
+    if state is not None:
+        silences = [
+            silence for silence in response
+            if silence['status']['state'] == state
+        ]
+    else:
+        silences = response
```

Comment:
Could this be a one-liner ??
`silences = [silence for silence in response if silence['status']['state'] == state] if state else response`

Reply:
It could be, but I don't see the point in doing this, other than making the code more compact and less readable.

Reply:
From the Zen of Python:

> Simple is better than complex.

and: 

> Sparse is better than dense.

Reply:
Well noted!